app/src/main/java/ch/epfl/sweng/eventmanager/ui/eventShowcase/models/ScheduleViewModel.java

```diff
@@ -43,16 +42,46 @@ public void init(int eventId) {
         this.eventId = eventId;
         this.scheduledItems = repository.getScheduledItems(eventId);
         this.joinedItems = buildJoinedScheduledItemsList(joinedScheduleItemRepository.findByEventId(eventId));
+        this.scheduleItemsByRoom = buildScheduleItemByRoom();
     }
 
     public LiveData<List<ScheduledItem>> getScheduledItems() {
         return scheduledItems;
     }
 
+
+    public LiveData<List<ScheduledItem>> getScheduledItemsForRoom(String room) {
+        return Transformations.map(getScheduleItemsByRoom(), map -> {
+            if (map == null) {
+                return null;
+            } else {
+                return map.get(room);
+            }
+        });
+    }
+
+    public LiveData<Map<String, List<ScheduledItem>>> getScheduleItemsByRoom() {
+        return scheduleItemsByRoom;
+    }
+
     public LiveData<List<ScheduledItem>> getJoinedScheduleItems() {
         return joinedItems;
     }
 
+    public Boolean isItemJoined(UUID itemId) {
+        // This method doesn't return a live-data because it's not watched by anything
+        // we only need to get a value at some point, and LiveDatas don't suit this model
+        List<ScheduledItem> events = getJoinedScheduleItems().getValue();
+        if (events != null) {
+            for (ScheduledItem i : events)
```

Comment:
Be coherent and use curly brackets everywhere! Even when they only take one line.